### PR TITLE
[i2c] Add target FSM responses to timeouts and read commands

### PR DESCRIPTION
**Pull request by a-will**
Change target FSM's stretch timeout to count across an entire transaction, for SMBus support.

Add a CSR to report fault events that occurred during a read, and gate
the next read (stretch clock at the first bit) until the event latch is
cleared. This is for reads to this IP's target module.

The gate helps prevent stale data from emerging from the FIFO.

Also roll in some of the functionality the TXRST_ON_COND CSR was meant
to provide, and make this a catch-all gate for releasing the TX FIFO on
a read instead. The prior mechanism had some corner cases that produced
races with software, such as software being able to inadvertently write
a new (yet stale) byte of data after the clear yet before being notified
that the command completed. The separate gating CSR helps avoid such
races by not allowing software's normal write path to bypass event
handling.

### CODE REVIEW

File: hw/ip/i2c/rtl/i2c_target_fsm.sv [hunk 1]
```diff
@@ -627,7 +635,7 @@ module i2c_target_fsm import i2c_pkg::*;
   //
```
Comment (hcallahan-lowrisc) [inline hw/ip/i2c/rtl/i2c_target_fsm.sv:635]: nit: stale comment (L631-L637)

File: hw/ip/i2c/data/i2c.hjson [hunk 2]
```diff
@@ -291,6 +291,20 @@
                 When 0, the bus monitor will report that the bus is always free, so the controller FSM is never blocked from transmitting.
                 '''
         }
+        { bits: "6"
+          resval: "0"
+          name: "TX_STRETCH_CTRL_EN"
+          desc: '''
+                Enable setting !!TARGET_EVENTS.TX_PENDING to gate target read transfers.
```
Comment (hcallahan-lowrisc) [inline hw/ip/i2c/data/i2c.hjson:298]: nit: imo this first line of the description is a bit too terse, I had to think quite hard to parse it the first time reading. L300 would be a better first line for the description I think.

**Review by hcallahan-lowrisc — APPROVED** [review]
LGTM, just minor doc nits that can be wrapped up into another PR if desired.

File: hw/ip/i2c/rtl/i2c_target_fsm.sv (hunk 1 above)
Comment (a-will) [reply hw/ip/i2c/rtl/i2c_target_fsm.sv:638]: Added info about the new term in the comment.

File: hw/ip/i2c/data/i2c.hjson (hunk 2 above)
Comment (a-will) [reply hw/ip/i2c/data/i2c.hjson]: Works for me. I deleted this line and made the one you selected be first.

**Review by andreaskurth — APPROVED** [review]
LGTM, thx @a-will!

**Comment by andreaskurth** [comment]
You ran this quickly through block-level DV to ensure that existing tests don't fall apart (pass rates roughly maintained), right?

**Comment by a-will** [comment]
> You did run this quickly through block-level DV to ensure that existing tests don't fall apart (pass rates roughly maintained), right?

Always. My process has been to fix DV tests first, then ensure they continue to pass--Otherwise, I make more fixes.
